### DISCUSSION TITLE
libc++ in Wolfi?

### DISCUSSION THREAD

**Discussion by isker** [wolfi-dev/os#4794] [Q&A] [answered]
Hi.  I'm trying out melange/Wolfi for the first time and I'm interested in building a project (https://github.com/cloudflare/workerd) that requires libc++; libstdc++ is not acceptable.

As far as I can tell, libc++ is not packaged in Wolfi, i.e. the gcc package has a libstdc++ subpackage, but the llvm package does not have a libc++ subpackage (if that's even how you'd distribute libc++; I'm not certain).

Is this an intentional exclusion?  I'm not very familiar with native toolchains, so I don't know what kinds of tradeoffs are involved here.

Thanks!

**Comment by dlorenc** [comment]
In general if we don't have a package that's needed, we can add it!

**Reply by isker** [reply]
Cool.  Maybe @kaniini (who has touched llvm15 a lot) could comment on the prospects of this.  Reading https://libcxx.llvm.org/BuildingLibcxx.html and trying to compare it to https://github.com/wolfi-dev/os/blob/main/llvm15.yaml, I am feeling pretty far in over my head 🌞.

**Reply by dlorenc** [reply]
I almost have it building!

**Reply by dlorenc** [reply, marked as answer]
https://github.com/wolfi-dev/os/pull/4852

**Reply by kaniini** [reply]
libc++ and libstdc++ are not compatible with each other.  We can't really support both at the same time, and libstdc++ is the one that is required by the LSB.

**Reply by kaniini** [reply]
Oh, apparently there is now a mode which allows libc++ to be built where it just extends libstdc++ with the extra functionality.  Maybe we can make that work.

**Reply by isker** [reply]
If it helps, workerd statically links libc++: https://github.com/cloudflare/workerd/pull/793

**Reply by isker** [reply]
Thank you for your help! I am away for a while but will try it out when I’m back.

**Reply by isker** [reply]
Well, I haven't been able to get a workerd build to succeed in melange.  But I have learned a lot in the process 🌞.  Here's the current situation: https://github.com/wolfi-dev/os/pull/5122.
